puppeteer/spec/request-modification.js

```diff
@@ -0,0 +1,104 @@
+const puppeteer = require("puppeteer");
+const userListSelector = "[data-id='users']";
+const {
+  BASE_URL,
+  USER_LIST_PATH,
+  SINGLE_USER_PATH,
+} = require("../data/constants.json");
+const chai = require("chai");
+const chaiAsPromised = require("chai-as-promised");
+const chaiDeepMatch = require("chai-deep-match");
+const expect = chai.expect;
+const expectedResponse = require("../data/response.json");
+chai.use(chaiAsPromised);
+chai.use(chaiDeepMatch);
+chai.use(require("chai-json-schema"));
+
+describe("Request Modification", () => {
+  let browser, page, interceptedRequest;
+  before(async function () {
+    browser = await puppeteer.launch();
+  });
+  beforeEach(async () => {
+    page = await browser.newPage();
+    await page.setCacheEnabled(false);
+    await page.goto(BASE_URL);
+    await page.setRequestInterception(true);
+  });
+  describe("Modified request verification", () => {
+    it("should be a GET method", async () => {
+      page.on("request", (request) => {
+        if (request.url() === `${BASE_URL}${USER_LIST_PATH}`) {
+          interceptedRequest = request;
+        }
+        request.continue();
+      });
+      page.click(userListSelector);
+      await page.waitForResponse(
+        (response) => response.url() === `${BASE_URL}${USER_LIST_PATH}`
+      );
+      expect(interceptedRequest.method()).to.be.equal("GET");
+    });
+
+    it("should have the modified URL", async () => {
```

Comment:
I would suggest adding some spacing and comments to make your tests more easy to read.